### ISSUE TITLE
[Bug] Cabinet teleports to doctor

### ISSUE THREAD

**Issue by Alberth289346** [CorsixTH/CorsixTH#2021]
#### Describe the issue
File cabinet moves by itself

#### Steps to Reproduce
1. Load save game attached to #2020 

2. Watch doctor in GP using the file cabinet

3. Cabinet teleports to doctor.

![Screenshot from 2021-07-18 14-24-01](https://user-images.githubusercontent.com/3254232/126067822-bd7125e7-d810-45af-9851-e9a564b62cbb.png)

![Screenshot from 2021-07-18 14-24-54](https://user-images.githubusercontent.com/3254232/126067828-fda28676-0fe6-43ca-b9ff-6e89c899eee6.png)

**Comment by mugmuggy** [comment]
This is a blocked off area bug. The walk doesn't actual walk anywhere, and the use animation just starts for the cabinet. Link to #826. I made note in #1462 of the drinks machine teleporting and its one a few other issues, (see #1682). There is another scenario not in #1682 where a patient teleports to the bench and off it.

However I think this scenario more closely relates to incorrectly blocking off areas vs the starting an action that shouldn't/can't start. TH itself for most blocked off areas doesn't check paths before a room is placed, place a room in a blocked off area fails. And with objects in a room it generally checks path to the door, but I don't think path checking occurs for fire extinguisher, bin, radiator or plant I can double check this when I get time. In the corridors its also possible to block an area off and that is more than likely due to the algorithm in place and it gives up.

**Comment by lewri** [comment]
Yes, leaning towards putting blocked off checks back inside rooms again. It seems to be causing more issues than just blocking off parts of the hospital.

**Comment by lewri** [comment]
I found this one by accident. Though... the doctor teleported to cabinet then back again, rather than the cabinet moving. Same cause though.

The doctor always sits down from the right of the chair. They also always will get up from the left. As mugmuggy said, this action of getting up/sitting down doesn't really move the doctor anywhere until after the fact. It's an odd design to force such movement, but alas we must work with it as so.

This bug also will occur if you put the desk so the chair isn't reachable from the right -- it actually causes a broken walk action and crashes the game.

**Comment by ARGAMX** [comment]
What is the expected solution to this issue?

Blocking off areas without executing the necessary checks that were present in the original TH is must not be allowed. 
Simply put, placing a table like this is should not be allowed.

Such placement isn't allowed in the original game. 
Why is it allowed in CorsixTH under certain conditions?
This also raises the question: why is the table placed this way in this save?
Obviously, the `allow_blocking_off_areas = true` was set to create this situation. 
But this is completely different from how it works in the original game.

In the original TH, you can't place objects in that way.
Thus, `allow_blocking_off_areas = true` emulates a non-existent invalid behavior.
I believe the `allow_blocking_off_areas = true` logic should be removed from CorsixTH.
Because it creates these kinds of unsolvable situations. It's not consistent with the original TH.
Then such defects will simply cease to exist, and therefore there will be nothing to fix.

Next, the placement mechanism should be implemented as in the original game, which I described here https://github.com/CorsixTH/CorsixTH/issues/826#issuecomment-3214374270

Therefore, it's unclear what the target of this issue is and what the final outcome should be.

Original Theme Hospital:

<img width="549" height="409" alt="Image" src="https://github.com/user-attachments/assets/37a0e0a7-b3b3-475d-a1e3-f5ed5b78e75b" />
